### DISCUSSION TITLE
js get JavascriptObjectRepository.Register

### DISCUSSION THREAD

**Discussion by 563256676** [cefsharp/CefSharp#4132] [General]
I used JavascriptObjectRepository.Register("key",new object()) to register an object, I can use await key.getSettings() on the webpage to get the registered Object, but I don't want to use asynchronous await to get it on the webpage , I want to get it in a synchronous way. Is there any way?

**Comment by amaitland** [comment]
What version of .Net are you targeting?

**Reply by 563256676** [reply]
I use CefSharp.Wpf.NETCore 101.0.180

**Reply by amaitland** [reply]
The NETCore package only supports async binding. 

Use await or then

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises
